### PR TITLE
[WIP] PaymentIntent and other experimentation. DO NOT MERGE!

### PR DESCRIPTION
**Pull request by thorsten-stripe**
to run, checkout this branch
run `npm install` to update Stripe node library
make sure webhooks are set up
run `npm run start`

### CODE REVIEW

File: public/javascripts/payments.js [hunk 1]
```diff
@@ -315,220 +303,214 @@
...
-   * This means there is a `source` parameter in the URL, and an active order.
-   * When this happens, we'll monitor the status of the order and present real-time
+   * This means there is a `source` parameter in the URL, and an active PaymentIntent.
+   * When this happens, we'll monitor the status of the PaymentIntent and present real-time
    * information to the user.
    */
 
-  const pollOrderStatus = async (
-    orderId,
+  const pollPaymentIntentStatus = async (
```
Comment (adreyfus-stripe) [inline public/javascripts/payments.js]: What are we polling this payment intent for? It's being called for sources that don't use PaymentIntents so I'm a little unclear how this works into those source flows (I haven't looked much at how sources interact with PaymentIntents)

Comment (tschaeff) [reply public/javascripts/payments.js]: For source types that aren't yet supported by PI I'm listening to the `source.chargeable` webhook and upon receiving I confirm the PaymentIntent server-side: https://github.com/stripe/stripe-payments-demo/pull/31/files?utf8=%E2%9C%93&diff=split#diff-e096badd84d65c3fbe372dced8268d69R93

Therefore I need to poll the PI on the client-side to know when it's status changed.

Comment (thorsten-stripe) [reply public/javascripts/payments.js]: For source types that aren't yet supported by PI I'm handling the source activation myself (handleSourceActivation()) and then I'm listening to the `source.chargeable` webhook. Upon receiving it, I confirm the PaymentIntent server-side: https://github.com/stripe/stripe-payments-demo/pull/31/files?utf8=%E2%9C%93&diff=split#diff-e096badd84d65c3fbe372dced8268d69R93

Therefore I need to poll the PI on the client-side to know when its status changed.

File: public/javascripts/store.js [hunk 2]
```diff
@@ -126,26 +95,26 @@ class Store {
...
+  setActivePaymentIntent(paymentIntent_client_secret) {
+    localStorage.setItem('paymentIntent_client_secret', paymentIntent_client_secret);
   }
 
-  // Get the active order ID from the local storage.
-  getActiveOrderId() {
-    return localStorage.getItem('orderId');
+  // Get the active PaymentIntent client secret from the local storage.
+  getActivePaymentIntent() {
+    return localStorage.getItem('paymentIntent_client_secret');
```
[the remark this answers is not in the document]
Comment (thorsten-stripe) [reply public/javascripts/store.js]: Ah yes, likely worth reviewing. However, I'm not sure whether sessionStorage stays intact throughout the redirects on all browsers.

Once PI supports all payment methods we will be able to get rid of this, as then the PI and its client_secret are appended to the `return_url` and we can then simply read it from there 👍

File: public/javascripts/payments.js [hunk 3]
```diff
@@ -158,51 +186,14 @@
...
       );
-      if (error) {
-        event.complete('fail');
-        await handleOrder({metadata: {status: 'failed'}}, null, error);
-      } else {
-        event.complete('success');
-        if (paymentIntent.status === 'succeeded') {
-          // No authentication required, show success message.
-          await handleOrder({metadata: {status: 'paid'}}, null, null);
-        } else if (paymentIntent.status === 'requires_source_action') {
```
Comment (thorsten-stripe) [inline public/javascripts/payments.js]: We need to think about how to best support 3D Secure with PRAPI.

**Pull request closed without merging**